controlled_navigation/controlled_navigation.py

```diff
@@ -59,31 +117,170 @@ def author_view(self, context):
 
         return fragment
 
-    def student_view(self, context):
+    def student_view(self, context: dict) -> Fragment:
         """
-        View for students to see the blocks in the unit.
+        View for students to navigate through the children.
+
+        Each child will be rendered and the student can navigate through them.
+
+        Args:
+            context (dict): Context for the student view.
+
+        Returns:
+            Fragment: The fragment with the child content.
         """
-        children_contents = []
         fragment = Fragment()
 
-        for child_id in self.children:
-            child = self.runtime.get_block(child_id)
-            child_fragment = self._render_child_fragment(child, context, "student_view")
-            fragment.add_fragment_resources(child_fragment)
-            children_contents.append(child_fragment.content)
+        children_ids = self.get_children_ids()
+
+        if not children_ids:
+            return fragment
+
+        child_usage_key = UsageKey.from_string(self.get_current_child_id(children_ids))
+        child = self.runtime.get_block(child_usage_key)
+        child_fragment = self._render_child_fragment(child, context, "student_view")
+        fragment.add_fragment_resources(child_fragment)
 
         render_context = {
             "block": self,
-            "children_contents": children_contents,
+            "child_content": child_fragment.content,
+            "is_first_child": self.is_first_child(children_ids),
+            "is_last_child": self.is_last_child(children_ids),
+            **context,
         }
-        render_context.update(context)
+
         fragment.add_content(
             LOCAL_RESOURCE_LOADER.render_django_template(
-                "static/html/children.html", render_context
+                "static/html/controlled_navigation.html",
+                render_context,
+                i18n_service=self.runtime.service(self, "i18n"),
             )
         )
+        fragment.add_javascript(
+            self.resource_string("static/js/src/controlled_navigation.js")
+        )
+        fragment.add_css(self.resource_string("static/css/controlled_navigation.css"))
+        fragment.initialize_js("XBlockControlledNavigation")
+
         return fragment
 
+    def get_current_child_id(self, children_ids: Optional[List[str]] = None) -> str:
+        """
+        Return the current child id.
+
+        Optionally receive a list of children to use.
+        If not provided, it will get the children from the block.
+
+        Args:
+            children_ids (Optional[List[str]], optional):
+                Children of the block. Defaults to None.
+
+        Returns:
+            str: The current child id.
+        """
+        if not self.current_child_id:
+            children = children_ids or self.get_children_ids()
+            self.current_child_id = children[0]
```

Comment:
does this fail if the children_ids list is empty?

Reply:
Yes, it would fail. However, the function is never called in case there are no children. Still, I will include a validation

Reply:
Done: https://github.com/eduNEXT/xblock-controlled-navigation/commit/3816dcf97e04141542f99d2a3f73b967e7e9ddd5